modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
 		forge_temperature += 5
 
 	if(forge_temperature > 0)
-		if(primitive)
-			icon_state = "primitive_forge_full"
-		else
-			icon_state = "forge_full"
+		icon_state = "[primitive ? "primitive_" : ""]forge_full"
 		light_range = 3
 	else if(forge_temperature <= 0)
-		if(primitive)
-			icon_state = "primitive_forge_empty"
-		else
-			icon_state = "forge_empty"
+		icon_state = "[primitive ? "primitive_" : ""]forge_empty"
 		light_range = 0
 
+/**
+ * Here we fix any weird in_use bugs
+ */
 /obj/structure/reagent_forge/proc/check_in_use()
 	if(!in_use)
 		return
 	for(var/mob/living/living_mob in range(1,src))
 		if(!living_mob)
 			in_use = FALSE
 
+/**
+ * Here we spawn coal depending on a chance of using wood
+ */
 /obj/structure/reagent_forge/proc/spawn_coal()
 	new /obj/item/stack/sheet/mineral/coal(get_turf(src))
 
 /obj/structure/reagent_forge/process()
-	if(world_check >= world.time) //to make it not too intensive, every 5 seconds
+	if(!COOLDOWN_FINISHED(src, forging_cooldown)) //every 5 seconds to not be too intensive, also balanced around 5 seconds
 		return
-	world_check += 5 SECONDS
+	COOLDOWN_START(src, forging_cooldown, 5 SECONDS)
 	check_fuel()
 	check_temp()
 	check_in_use() //plenty of weird bugs, this should hopefully fix the in_use bugs
 
+/obj/structure/reagent_forge/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	var/user_smithing_skill = user.mind.get_skill_level(/datum/skill/smithing)
+	switch(user_smithing_skill)
+		if(SKILL_LEVEL_NONE to SKILL_LEVEL_NOVICE)
+			to_chat(user, span_notice("[src] requires you to be more experienced!"))
+		if(SKILL_LEVEL_APPRENTICE to SKILL_LEVEL_JOURNEYMAN)
+			if(forge_level >= FORGE_LEVEL_ONE)
+				to_chat(user, span_warning("Your level of smithing has already influenced [src]!"))
+				return
+			to_chat(user, span_notice("[src] levels up... examine to view what has changed!"))
```

Comment:
```suggestion
			to_chat(user, span_notice("As you work with [src], you note the purity caused by heating metal with nothing but exposed flame.")
```
This makes the ore improvement seem like a realization rather than a blessing; LEARNED rather than GIVEN